notebook/04_portfolio_trade.ipynb

```diff
@@ -4,378 +4,337 @@
     "\n",
-    "import warnings\n",
-    "from gquant.dataframe_flow import TaskGraph, Node\n",
-    "import nxpd\n",
-    "import ipywidgets as widgets\n",
-    "from nxpd import draw\n",
-    "import os\n",
+    "It is very time-consuming for CPUs to crunch massive amount of data and compute the complicated data science models. Large data set requires distributed computation, which is too complicated for data scientist to adopt.\n",
     "\n",
-    "warnings.simplefilter(\"ignore\")"
-   ]
-  },
-  {
-   "cell_type": "markdown",
-   "metadata": {},
-   "source": [
-    "Start the Dask local cluster environment for distrubuted computation:"
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": 2,
-   "metadata": {},
-   "outputs": [
-    {
-     "data": {
-      "text/html": [
-       "<table style=\"border: 2px solid white;\">\n",
-       "<tr>\n",
-       "<td style=\"vertical-align: top; border: 0px solid white\">\n",
-       "<h3>Client</h3>\n",
-       "<ul>\n",
-       "  <li><b>Scheduler: </b>tcp://127.0.0.1:43772\n",
-       "  <li><b>Dashboard: </b><a href='http://127.0.0.1:8787/status' target='_blank'>http://127.0.0.1:8787/status</a>\n",
-       "</ul>\n",
-       "</td>\n",
-       "<td style=\"vertical-align: top; border: 0px solid white\">\n",
-       "<h3>Cluster</h3>\n",
-       "<ul>\n",
-       "  <li><b>Workers: </b>8</li>\n",
-       "  <li><b>Cores: </b>8</li>\n",
-       "  <li><b>Memory: </b>536.39 GB</li>\n",
-       "</ul>\n",
-       "</td>\n",
-       "</tr>\n",
-       "</table>"
-      ],
-      "text/plain": [
-       "<Client: scheduler='tcp://127.0.0.1:43772' processes=8 cores=8>"
-      ]
-     },
-     "execution_count": 2,
-     "metadata": {},
-     "output_type": "execute_result"
-    }
-   ],
-   "source": [
-    "from dask_cuda import LocalCUDACluster\n",
-    "cluster = LocalCUDACluster()\n",
-    "from dask.distributed import Client\n",
-    "client = Client(cluster)\n",
-    "client"
+    "As a consequence, the quantitative workflow has become more complicated than ever. It integrates massive data from different sources, requiring multiple iterations to obtain significative results. \n",
+    "\n",
+    "**gQuant** has been developed to address all these challenges by organizing dataframes into graphs. It introduces the idea of **dataframe-flow**, which manipulates dataframes at graph level. An **acyclic directed graph** is defined, where the nodes are dataframe processors and the edges are the directions of passing resulting dataframes.\n",
+    "\n",
+    "With a graph approach, quant's workflow is described at a high level, letting quant analysts address the complicated workflow challenge.\n",
+    "\n",
+    "Built on top of [RAPIDS.ai](https://rapids.ai) library, gQuant manipulates and stores the data in GPU and GPU memory.\n",
```

Comment:
" stores the data in GPU and GPU memory." -> " stores the data in GPU memory."

Reply:
Done